flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/catalog/hive/HiveCatalogGenericMetadataTest.java

```diff
@@ -317,6 +319,21 @@ public void testFunctionCompatibility() throws Exception {
         assertEquals(FunctionLanguage.JAVA, catalogFunction.getFunctionLanguage());
     }
 
+    @Test
+    public void testGenericTableWithoutConnectorProp() throws Exception {
+        catalog.createDatabase(db1, createDb(), false);
+        TableSchema tableSchema =
+                TableSchema.builder()
+                        .fields(
+                                new String[] {"s", "ts"},
+                                new DataType[] {DataTypes.STRING(), DataTypes.TIMESTAMP_LTZ(3)})
+                        .watermark("ts", "ts-INTERVAL '1' SECOND", DataTypes.TIMESTAMP_LTZ(3))
+                        .build();
+        CatalogTable catalogTable = new CatalogTableImpl(tableSchema, Collections.emptyMap(), null);
+        catalog.createTable(path1, catalogTable, false);
+        assertEquals(tableSchema, catalog.getTable(path1).getSchema());
```

Comment:
Should we assert the table's property IS_GENERIC is true in here as for such table without 'connector' or 'connector.type', this pr will put IS_GENERIC = true into it's properties .

Reply:
No, the `IS_GENERIC` property should always be removed upon retrieval. I'll update the test to verify that.